### DISCUSSION TITLE
Field permissions based on separate item permissions

### DISCUSSION THREAD

**Discussion by Haffi112** [directus/directus#6099] [Archive]
I have a table called "Project" and within it, I have an M2O field called "Owner" and an M2M field called "Participants". Some of the fields in the "Project" table are only relevant to the "Owner" of the project but others are relevant to all of the participants. Both the owner and the participants have the same role.

When setting up read access for the table I cannot give the participants less access to fields of the table than the owner.

Is the current solution to create a new table called "Project Options" and only give the owner access to that? Or is it possible to have two filters and field permissions depending on each filter? (Currently, I don't see a way to do that but it would definitely make my life easier).

**Comment by benhaynes** [comment]
Interesting! Our filter rules are for the action of reading the item, but those rules are separate from the field level control. I wouldn't want to make the permissions configuration overly complex, but I wonder if it's worth considering adding rules to the field level permissions.

Alternatively, and perhaps more appropriately, this might be able to be handled by conditional fields... which is a sponsored feature that will be implemented in the coming weeks.

**Reply by Haffi112** [reply]
Thank you for the quick response, Ben. I agree that making this accessible is a UX challenge.

Conditional fields are then perhaps the right way to solve this in a clean manner (I didn't know about them). For a good UX, it would be nice to be able to apply the same filter to several fields simultaneously such that editing the filters is a breeze.

**Reply by benhaynes** [reply]
Agree. We have an issue open to track Conditional Fields... so I won't duplicate our thinking there. But the goal is to make it very flexible but still allow easy configuration (the goal of all Directus).
